src/condor_tools/history.cpp

```diff
@@ -1387,6 +1220,78 @@ static void printJobAds(ClassAdList & jobs)
 	jobs.Close();
 }
 
+static bool isvalidattrchar(char ch) { return isalnum(ch) || ch == '_'; }
+
+/*
+*	Function to parse banner information of both history and epoch files (They vary slightly)
+*	and fill info into Banner info struct. After parsing banner line we will then determine
+*	if we are only matching for Cluster, Cluster.Proc, or Owner then check if banner info
+*	matches.
+*/
+static bool parseBanner(BannerInfo& info, std::string banner) {
+	//Parse Banner info
+	int clust = -1, proc = -1;
+
+	const char * p = banner.c_str();
+	const char * endp = p + banner.size();
+
+	classad::ClassAdParser parser;
+	parser.SetOldClassAd(true);
+
+	//fprintf(stdout, "parseBanner(%s)\n", p);
+	while (*p == '*') ++p;
+	while (isspace(*p)) ++p;
+
+	bool parsedFully = true;
```

Comment:
if the banner is empty, parsedFully ends up being true,   is that really what you want?